flag/flags.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/spf13/pflag"
+	"k8s.io/klog"
+
+	utilnet "k8s.io/apimachinery/pkg/util/net"
```

Comment:
thanks for the PR, but things in this repo cannot depend on other Kubernetes repos (see https://github.com/kubernetes/utils#criteria-for-adding-code-here)

Reply:
Thanks for the tip. Will modify for sure. Sorry I thought a draft PR would not bother others. Should I close this and reopen when everything is ready or it is ok to keep it here?

Reply:
it's not clear to me that the dependencies that come along with this are ones we want in k8s.io/utils... was there an issue that prompted this move?

Reply:
This is part of this [issue](https://github.com/kubernetes/cloud-provider/issues/29). I will close this pr and move the dependency to somewhere else. Thanks